src/cpp/session/modules/SessionDataImportV2.R

```diff
@@ -13,21 +13,162 @@
 #
 #
 
-.rs.addJsonRpcHandler("preview_data_import", function(path, maxRows, maxCols = 100, maxFactors = 64)
+.rs.addFunction("assemble_data_import_name", function(dataImportOptions)
+{
+  dataName <- dataImportOptions$dataName
+
+  if (is.null(dataName) || dataName == "")
+    {
+      if (!is.null(dataImportOptions$importLocation) && dataImportOptions$importLocation != "")
+      {
+        basename <- basename(dataImportOptions$importLocation)
+        if (length(basename) > 0)
+        {
+          fileComponents <- unlist(strsplit(basename, ".", fixed = TRUE))
+          components <- length(fileComponents)
+          if (components >= 1)
+          {
+            dataName <- paste(fileComponents[1:components-1])
+          }
+        }
+      }
+    }
+  
+  dataName
+})
+
+.rs.addFunction("assemble_data_import_parameters_csv", function(dataImportOptions)
+{
+  parameters <- ""
+  spacing <- ",\n  "
+  useCsv <- (!is.null(dataImportOptions$delimiter) && dataImportOptions$delimiter == ",")
+
+  if (!is.null(dataImportOptions$delimiter) && dataImportOptions$delimiter != ",")
+  {
+    parameters <- paste0(parameters, spacing, "delim = \"", dataImportOptions$delimiter, "\"")
+  }
+
+  if (!is.null(dataImportOptions$quotes) && dataImportOptions$quotes != "" && !useCsv)
+  {
+    parameters <- paste0(parameters, spacing, "quote = \"", dataImportOptions$quotes, "\"")
+  }
+
+  if (!is.null(dataImportOptions$escapeBackslash) && !useCsv)
```

Comment:
You could make this more declarative (and not dependent on package defaults) by making an R list of parameter values, and then diffing the list with `formals(readr::read_csv)` and `formals(readr::read_delim)`--pick whichever has a shorter diff and use the diff as the argument list.